### PR TITLE
HPCC-27830 Azure LogAccess use Secrets

### PR DESCRIPTION
**Pull request by rpastrana**
- Adds esp jsecret category
- Eliminates use of Env vars
- Stablishes azure_logaccess secret
- Updates documentation
- Provides secrets template

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: helm/examples/azure/log-analytics/README.md [hunk 1]
```diff
@@ -6,37 +6,53 @@ a Log Analytics workspace, and explicitly enabling the Log Analytics feature. A
...
+```
 
-This means functionality which fetches HPCC component logs will attempt to gather logs via KQL queries.
+#### c - Configure HPCC logAccess
+The target HPCC deployment should be directed to target the above Azure Log Analytics workspace by providing appropriate logAccess values (such as ./loganalytics-hpcc-logaccess.yaml). The previously created azure-logaccess secret must be declared and associated with the esp category, this can be accomplished via secrets value yaml (such as ./loganalytics-logccess-secrets.yaml)
 
 Example use:
 ```console
-  helm install myhpcc hpcc/hpcc -f HPCC-Platform/helm/examples/azure/log-analytics/loganalytics-hpcc-logaccess.yaml
+  helm install myhpcc hpcc/hpcc -f HPCC-Platform/helm/examples/azure/log-analytics/loganalytics-hpcc-logaccess.yaml -f  HPCC-Platform/helm/examples/azure/log-analytics/loganalytics-logccess-secrets.yaml
```
Comment (rpastrana) [inline helm/examples/azure/log-analytics/README.md:56]: @afishbeck can you confirm this is the intended approach for declaring jsecret secrets?
I'm concerned the secrets/vaults structure might be lead to inadvertent overwrites of a given secret/vault category (in this case 'esp')

**Comment by rpastrana** [comment]
@afishbeck please review changes from your review. 
Note, I added a helper script to create the required secret.

File: system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp [hunk 2]
```diff
@@ -253,43 +253,55 @@ static void submitKQLQuery(std::string & readBuffer, const char * token, const c
...
     {
-        WARNLOG("%s: Environment variable 'AZURE_CLIENT_ID' not found!", COMPONENT_NAME);
-        m_clientID.set(logAccessPluginConfig.queryProp("connection/@clientID"));
+        WARNLOG("%s: Could not find '%s.%s' secret value!", COMPONENT_NAME, azureLogAccessSecretName, azureLogAccessSecretAADClientID);
+        m_aadClientID.set(logAccessPluginConfig.queryProp("connection/@clientID"));
 
-        if (m_clientID.isEmpty())
-           throw makeStringException(-1, "Could not find Azure AD client ID, set 'AZURE_CLIENT_ID' env var, or connection/@clientID in AzureClient LogAccess configuration - format is '00000000-0000-0000-0000-000000000000'!");
+        if (m_aadClientID.isEmpty())
+          throw makeStringExceptionV(-1, "%s: Could not find AAD Client ID, provide it as part of %s.%s secret, or connection/@clientID in AzureClient LogAccess configuration!", COMPONENT_NAME, azureLogAccessSecretName, azureLogAccessSecretAADClientID);
```
Comment (afishbeck) [inline system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp:288]: Some of the indentations in this file look a bit off.

File: helm/examples/azure/log-analytics/create-azure-logaccess-secret.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+if [[ $? -ne 0 ]]
+then
+  echo "Error creating: Target secret '${secretname}'!"
+  echo >&2
+  usage
+  exit 1
+else
+  echo "Target secret '${secretname}' successfully created!"
+  ${k8scommand} get secret ${secretname}
+fi
```
Comment (afishbeck) [inline helm/examples/azure/log-analytics/create-azure-logaccess-secret.sh:66]: Git complains about these files without newlines at the end.  For bash and yaml I would add newlines.  For the secret content files you can't.

**Review by afishbeck — APPROVED** [review]
@rpastrana a few trivial comment, but looks good.

**Comment by rpastrana** [comment]
@afishbeck integrated the minor changes you suggested

**Comment by rpastrana** [comment]
@ghalliday @richardkchapman ready for merge

**Review by ghalliday — APPROVED** [review]

**Pull request merged**